jplag.endToEndTesting/src/test/java/de/jplag/end_to_end_testing/JPlagEndToEndTestingSuiteTest.java

```diff
@@ -0,0 +1,258 @@
+package de.jplag.end_to_end_testing;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+
+import de.jplag.JPlag;
+import de.jplag.JPlagComparison;
+import de.jplag.JPlagResult;
+import de.jplag.end_to_end_testing.helper.FileHelper;
+import de.jplag.end_to_end_testing.helper.JPlagTestSuiteHelper;
+import de.jplag.end_to_end_testing.helper.JsonHelper;
+import de.jplag.end_to_end_testing.model.ExpectedResult;
+import de.jplag.end_to_end_testing.model.Options;
+import de.jplag.end_to_end_testing.model.ResultDescription;
+import de.jplag.exceptions.ExitException;
+import de.jplag.options.JPlagOptions;
+import de.jplag.options.LanguageOption;
+
+/**
+ * Main test class for end-to-end testing in all language. The test cases aim to detect changes in the detection of
+ * plagiarism in the Java language and to be able to roughly categorize them. The plagiarism is compared with the
+ * original class. The results are compared with the results from previous tests and changes are detected.
+ */
+public class JPlagEndToEndTestingSuiteTest {
```

Comment:
Convoluted name, rename it to `EndToEndTestSuite` or similar.

Reply:
Without the test at the end, the classes don't run in the pipeline, do they?

Reply:
Good point, I am not sure. You can test that with a local maven run. If it is the case, maybe we just go for `EndToEndSuiteTest` for now.

Reply:
maybe `EndToEndSuiteTest` ?

Reply:
without the "test" at the end the tests don't run locally either. so i will rename it to` EndToEndSuiteTest`

Reply:
done